### DISCUSSION TITLE
Use case of plotting.TEAM_COLORS and plotting.TEAM_TRANSLATE

### DISCUSSION THREAD

**Discussion by Casper-Guo** [theOehrly/Fast-F1#417] [General]
In the laps dataframes, Red Bull is called Red Bull Racing and Haas is called Haas F1 team. Neither of these is in `plotting.TEAM_COLORS`

I have never encountered the three-letter team codes while using FastF1 so I also wonder what use case was `plotting.TEAM_TRANSLATE` designed for.

As seen in the [qualifying results example](https://docs.fastf1.dev/examples_gallery/plot_qualifying_results.html#sphx-glr-examples-gallery-plot-qualifying-results-py), the only way to get team color from team name is by using `plotting.team_color`. Contrast this against the drivers, where `DRIVER_COLORS` and `DRIVER_TRANSLATE` are sufficient.

I know reworking the plotting module is being planned but maybe some improvements can be made to the team situation in the short term since the two dictionaries are very unwieldy/redundant currently.

**Comment by theOehrly** [comment]
The team names in the laps dataframe are taken from the F1 API. Those are the same names as used on formula1.com
 
![grafik](https://github.com/theOehrly/Fast-F1/assets/23384863/355129ec-7dab-43c4-bc15-b7c8bf894e39)
 
I have absolutely no idea, why the teams are named the way the are. Because those are not the official team names as far as I know.

The three letter team codes... that's a very good question. I have no idea to be honest. I'm pretty sure I haven't made them up. But I also can't tell you where I got them from or what they are used for. An interesting mystery 😅

Getting to the fact that the team names in the dictionary don't match the dataframe. This is because the team names in the dictionary are intentionally shortened to their most recognizable part so that the fuzzy search in ``.team_color`` works more reliably.

In terms of API consitency. Yes, you can argue that you can use ``DRIVER_COLORS`` and ``DRIVER_TRANSLATE`` but you can't do the same for teams. On the other hand, you could just use both ``.team_color`` and ``.driver_color`` in both cases and you have a consistent interface. In my opinion, the dictionaries themselves should not be public API. An exposed dictionary allows for very little flexibility in the underlying data and it gives very little room to change stuff without breaking existing functionality.

**Comment by Casper-Guo** [comment]
> An interesting mystery

Dug through the blame history for a minute to locate the original [commit](https://github.com/theOehrly/Fast-F1/commit/482d3ddc13da51e0c852f2dcf8094e1d318e3dbf#diff-5a2da04bd73eab3364f1e13665c5e0ad0d8eccc5f1d8176a9810573342219834),  looks like the entire file was added back then. No telling what the intended use case had been then.

**Reply by theOehrly** [reply]
Yes, I certainly didn't add it originally. But I'm sure I must have modified some entries of it. Or have team names not changed in the last 3-4 years? 
The entire ``plotting.py`` is a huge legacy thing and has never really changed.

**Reply by Casper-Guo** [reply]
Team names have been updated periodically

**Reply by theOehrly** [reply]
I checked the blame history as well now. What I meant was, I made the change from Toro Rosso (STR) to Alpha Tauri (APT). But I can't remember how or why I decided on  APT as the abbreviation.
